### PR TITLE
Update versionComparisonURL to check build versions

### PR DESCRIPTION
**Pull request by Jag96**
@roryabraham will you please review this?
cc @AndrewGable 

This PR updates `generateVersionComparisonURL` to check the build number in the version. I've also updated the tests to cover more version comparison cases. Once this is merged I'll update the expensify-common hashes in all the other repos.

### Fixed Issues
Related to https://github.com/Expensify/Expensify/issues/156562

# Tests
Added automated tests

# QA
N/A, we'll fully test this when we deploy

### CODE REVIEW

**Comment by roryabraham** [comment]
@Jag96 No need to update expensify-common git hashes. I'm actually going to end up [moving this code back over to Expensify.cash](https://github.com/Expensify/Expensify.cash/pull/1690), because it just doesn't need to be in expensify-common.

**Comment by Jag96** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/GithubUtils.js [hunk 1]
```diff
@@ -105,28 +105,29 @@ export default class GithubUtils {
...
-                        && semverSatisfies(repoTag, `<${tagSemver.major}.${tagSemver.minor}.x`)
+                        && semverSatisfies(repoTag, `<${tagSemver.major}.${tagSemver.minor}.x`, {includePrerelease: true})
                     ) {
                         resolve(getComparisonURL(repoTag, tagSemver));
                         return true;
                     }
 
                     if (semverLevel === 'PATCH'
-                        && semverSatisfies(repoTag, `<${tagSemver}`)
+                        && semverSatisfies(repoTag, `<${tagSemver}`, {includePrerelease: true,})
```
Comment (roryabraham) [inline lib/GithubUtils.js]:
```suggestion
                        && semverSatisfies(repoTag, `<${tagSemver}`, {includePrerelease: true})
```

**Review by roryabraham — CHANGES_REQUESTED** [review]
Code looks good, but can we also fix the max-len lint warnings too?

**Comment by Jag96** [comment]
@roryabraham updated!

**Review by roryabraham — APPROVED** [review]
Thanks!

**Pull request merged**